### PR TITLE
Pal 651 user service redis

### PR DESCRIPTION
**Pull request by dev930018**

### CODE REVIEW

**Review by pd104923 — DISMISSED** [review]
I've approved the code but I'm curious as to how the h2 db files got added to the repo. These should probably be removed.

File: charts/redis/templates/NOTES.txt
Comment (nw1984) [inline charts/redis/templates/NOTES.txt]: Changes in this area are also happening as part of PAL-683

**Review by nw1984 — DISMISSED** [review]
Need to wait for PAL-683 to be committed before committing these changes.

File: user-service/src/main/resources/application-redis.yaml
Comment (dev930018) [inline user-service/src/main/resources/application-redis.yaml:30]: Should we add this to the backlog?

Comment (developer01189998819991197253) [reply user-service/src/main/resources/application-redis.yaml:30]: @dev930018 whats the expected release date for Spring 2.4? is it in development/predicted?

File: Chart.yaml
Comment (ac74475) [inline Chart.yaml]: Why the change? if the change is necessary can we maintain consistency and therefore change the name as well

File: Chart.yaml
Comment (ac74475) [inline Chart.yaml]: same here

File: README.md
Comment (ac74475) [inline README.md]: The readme needs to explain the difference between redis and redis cluster

File: charts/kafka/charts/zookeeper/templates/zookeeper-pv.yaml
Comment (ac74475) [inline charts/kafka/charts/zookeeper/templates/zookeeper-pv.yaml:15]: surely you only want to install this PV if you are installing Kafka?

File: charts/kafka/templates/kafka-pv.yaml
Comment (ac74475) [inline charts/kafka/templates/kafka-pv.yaml:15]: again surely we only want to install this PV if kafka is to be installed

File: charts/redis-cluster/values.yaml
Comment (ac74475) [inline charts/redis-cluster/values.yaml]: can we use a specific tag

**Review by ac74475 — COMMENTED** [review]
Here are a few initial comments, I think @dev-958 should review this before it gets merged

File: Chart.yaml
Comment (dev930018) [reply Chart.yaml]: In kubernetes some things are treated as DNS names, which dont allow caps but could follow this naming convention. In go templates, they need to follow go variable naming conventions, where `metrics-server` is `metrics subtract server`... The camelCase is better choice in this case i believe (evidenced in the redisCluster vs redis-cluster index and pluck commands)

File: charts/kafka/charts/zookeeper/templates/zookeeper-pv.yaml
Comment (dev930018) [reply charts/kafka/charts/zookeeper/templates/zookeeper-pv.yaml:15]: This is already done with the Chart.yaml global.kafka.install condition, it was needed when the pv was part of the root-level templates, but underneath zookeeper templates it's not needed

File: user-service/src/main/resources/application-redis.yaml
Comment (dev930018) [reply user-service/src/main/resources/application-redis.yaml:30]: https://spring.io/blog/2020/06/29/spring-boot-2-4-0-m1-is-now-available
October-November?

Comment (ac74475) [reply user-service/src/main/resources/application-redis.yaml:30]: yes, add a ticket on the backlog

**Review by kg981167224 — APPROVED** [review]

File: user-service/src/main/resources/application-redis.yaml
Comment (dev930018) [reply user-service/src/main/resources/application-redis.yaml:30]: Raised PAL-734 with due-date October

**Review by pd104923 — APPROVED** [review]

**Review by dev-958 — APPROVED** [review]

**Review by ac74475 — APPROVED** [review]

**Pull request merged**
